### PR TITLE
fix unmounted warning, image size on sliced card, and srolling on que…

### PR DESCRIPTION
**Pull request by kamiranoff**
…stion

### CODE REVIEW

File: src/screens/Playground/NoMoreCards/NoMoreCards.tsx [hunk 1]
```diff
@@ -18,9 +17,8 @@ interface Props {
 const NoMoreCards: FC<Props> = ({ deckId }) => {
   const badAnswers = useSelector(selectBadAnswers(deckId));
   const goodAnswers = useSelector(selectGoodAnswers(deckId));
-  const scoreInfo = getUserScore(goodAnswers, badAnswers);
-  // @ts-ignore
-  const icon = animations[`${scoreInfo.icon}Lady`];
+  const scoreInfo = getLadyScoreInfo(goodAnswers, badAnswers);
+  const icon = scoreInfo.icon;
```
Comment (kamiranoff) [inline src/screens/Playground/NoMoreCards/NoMoreCards.tsx:21]: I had a crash here. When I tried to share, then signup with google after getting the result of the playground. So now it should be fixed.

File: src/screens/Playground/NoMoreCards/getUserScore.ts [hunk 2]
```diff
@@ -5,11 +5,11 @@ const getRandom = (items: { quote: string; author: string }[]) =>
 
 const getUserScore = (correct: number, incorrect: number) => {
   const total = correct + incorrect;
-  const score = Math.round((correct / total) * 100);
+  const score = Math.round(total ? (correct / total) * 100 : 0);
```
Comment (kamiranoff) [inline src/screens/Playground/NoMoreCards/getUserScore.ts:8]: Makes sure total is not 0 and score won't end up being Infinity

File: src/screens/Playground/NoMoreCards/getUserScore.ts [hunk 3]
```diff
@@ -5,11 +5,11 @@ const getRandom = (items: { quote: string; author: string }[]) =>
...
   const info = {
     primary: '',
     secondary: '',
-    icon: '',
+    icon: 'meh',
```
Comment (kamiranoff) [inline src/screens/Playground/NoMoreCards/getUserScore.ts:12]: I believe the crash came from here. If the score is not computed properly then the icon name was wrong and Lottie crashed.

File: src/screens/Playground/NoMoreCards/utils.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  primary: string;
+  secondary: string;
+  icon: keyof typeof animations;
+  score: string;
+};
+
+const getRandom = (items: { quote: string; author: string }[]) =>
+  items[Math.floor(Math.random() * items.length)];
+
+export const getLadyScoreInfo = (correct: number, incorrect: number): Info => {
```
Comment (kamiranoff) [inline src/screens/Playground/NoMoreCards/utils.ts:14]: this makes sure that the types are correct for the icon, and should avoid any unexpected crash.
This file might need to be rename/move somewhere more appropriate. Let me know

File: src/screens/Playground/NoMoreCards/NoMoreCards.tsx (hunk 1 above)
Comment (heron2014) [reply src/screens/Playground/NoMoreCards/NoMoreCards.tsx:21]: ohh intresting

File: src/screens/Playground/NoMoreCards/getUserScore.ts (hunk 2 above)
Comment (heron2014) [reply src/screens/Playground/NoMoreCards/getUserScore.ts:8]: great!!

**Review by heron2014 — APPROVED** [review]

**Pull request merged**